iceoryx_posh/include/iceoryx_posh/iceoryx_posh_types.hpp

```diff
@@ -192,15 +177,61 @@ struct DefaultChunkQueueConfig
 };
 
 // alias for cxx::string
-using ConfigFilePathString_t = cxx::string<1024>;
 using ProcessName_t = cxx::string<MAX_PROCESS_NAME_LENGTH>;
 using NodeName_t = cxx::string<100>;
 
+/// @todo Move everything in this namespace to iceoryx_roudi_types.hpp once we move RouDi to a separate CMake target
+namespace roudi
+{
+using ConfigFilePathString_t = cxx::string<1024>;
+
+constexpr char MQ_ROUDI_NAME[] = "/roudi";
+
+/// shared memmory segment for the iceoryx managment data
+constexpr char SHM_NAME[] = "/iceoryx_mgmt";
+
+// Timeout
+using namespace units::duration_literals;
+constexpr units::Duration PROCESS_DEFAULT_KILL_DELAY = 45_s;
+constexpr units::Duration PROCESS_TERMINATED_CHECK_INTERVAL = 250_ms;
+constexpr units::Duration PROCESS_WAITING_FOR_ROUDI_TIMEOUT = 60_s;
+constexpr units::Duration DISCOVERY_INTERVAL = 100_ms;
```

Comment:
the `PROCESS_KEEP_ALIVE_INTERVAL` depends on this, so the same as `MQ_ROUDI_NAME` and `SHM_NAME` applies to this

Reply:
Moved `PROCESS_WAITING_FOR_ROUDI_TIMEOUT` to namespace `runtime`. Also, I found the first circular dependency between RouDi and the runtime. roudi::DISCOVERY_INTERVAL -> runtime::PROCESS_KEEP_ALIVE_INTERVAL -> roudi::PROCESS_KEEP_ALIVE_TIMEOUT :rofl: We really need to untangle RouDi and the runtime and separate them clearly in CMake targets. Just wondering if this will be possible in v1.x without breaking the user API. @dkroenke Do we have a list how the folks integrating iceoryx are build it CMake-wise? Is it possible to introduce a separate `iceory_roudi` CMake package in v1.x?

Reply:
Yes we have an instruction here: [https://github.com/eclipse/iceoryx/blob/master/doc/installation-guide.md#build-with-cmake](https://github.com/eclipse/iceoryx/blob/master/doc/installation-guide.md#build-with-cmake) and a separate library for roudi called "iceoryx_posh_roudi" here: [here](https://github.com/eclipse/iceoryx/blob/3c43d88fb385747bdb00c5c48e6cca89053df8a3/iceoryx_posh/CMakeLists.txt#L201) Also the iox-roudi executable is defined in this file.

We can think of to move the roudi lib into an own cmake file.

Reply:
short term, you can create a constexpr in runtime and assign that to `roudi::DISCOVERY_INTERVAL` and `runtime::PROCESS_KEEP_ALIVE_INTERVAL` ... you just have to find a good name for it

Reply:
We should do that in a follow-up